### PR TITLE
Finalizing tasks: tests, ER diagram, cleanup

### PR DESCRIPTION
**Pull request by Dominic-dul**
-Separated player input validation from view folder to separate place for testability
-Created unit tests for register info validation
  --Also checks, if service is being called to add the validated player to the database
-Did some data binding in registration page
-Created ER diagram for our database
-Implemented extension method for ObservableRangeCollection in leaderboard so the players are now displayed in descending order (by points)
-Cleaned up unecessary buttons in leaderboard

### CODE REVIEW

**Review by MartynasAnn — APPROVED** [review]
Looks good, didn't find any possible issues!

File: EncounterMeApp/EncounterMeApp/PlayerManager.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                return null;
+            }
+            if (string.IsNullOrEmpty(newPlayer.Lastname))
+            {
+                return null;
+            }
+            if (string.IsNullOrEmpty(newPlayer.Password))
+            {
+                return null;
+            }
```
Comment (PijusPale) [inline EncounterMeApp/EncounterMeApp/PlayerManager.cs:41]: Seems like a repetetive code, there's got to be a way to simplify it

File: EncounterMeApp/EncounterMeApp/ViewModels/LeaderboardViewModel.cs [hunk 2]
```diff
@@ -30,16 +29,10 @@ public LeaderboardViewModel()
             //var image = "https://cdn3.iconfinder.com/data/icons/games-11/24/_user-512.png";
```
Comment (PijusPale) [inline EncounterMeApp/EncounterMeApp/ViewModels/LeaderboardViewModel.cs:29]: Will we need this in the future?

**Review by PijusPale — APPROVED** [review]
Great job!

**Pull request merged**